packages/zoe/src/contracts/callSpread/calculateShares.js

```diff
@@ -12,28 +14,22 @@ import { oneMinus, make100Percent, make0Percent } from './percent';
  * of the underlying asset at closing that determines the payouts to the parties
  *
  * @type {CalculateShares} */
-function calculateShares(
-  strikeMath,
-  collateralMath,
-  price,
-  strikePrice1,
-  strikePrice2,
-) {
-  const collateralBrand = collateralMath.getBrand();
-  if (strikeMath.isGTE(strikePrice1, price)) {
+function calculateShares(collateralBrand, price, strikePrice1, strikePrice2) {
+  if (amountMath.isGTE(strikePrice1, price)) {
     return {
       longShare: make0Percent(collateralBrand),
       shortShare: make100Percent(collateralBrand),
     };
-  } else if (strikeMath.isGTE(price, strikePrice2)) {
+  } else if (amountMath.isGTE(price, strikePrice2)) {
     return {
       longShare: make100Percent(collateralBrand),
       shortShare: make0Percent(collateralBrand),
     };
   }
 
-  const denominator = strikeMath.subtract(strikePrice2, strikePrice1);
-  const numerator = strikeMath.subtract(price, strikePrice1);
+  const denominator = amountMath.subtract(strikePrice2, strikePrice1);
+  const numerator = amountMath.subtract(price, strikePrice1);
+  assert.typeof(numerator.value, 'bigint');
   assert.typeof(denominator.value, 'bigint');
```

Comment:
Let me see if the type checking is satisfied by Nat. It may be

Reply:
Ok, just checked, and `isNat` isn't currently sufficient for the type checking. We can probably change it so that it is, but that would require a change on the Nat repo and seems out of scope for this PR. 

<img width="554" alt="Screen Shot 2021-03-16 at 11 08 04 AM" src="https://user-images.githubusercontent.com/2441069/111358817-02f04480-8648-11eb-8ae9-75fc253d9f2a.png">